ldms/src/core/ldms_xprt.c

```diff
@@ -3764,20 +3764,24 @@ int ldms_xprt_sockaddr(ldms_t _x, struct sockaddr *local_sa,
 {
 	struct ldms_xprt *x = _x;
 	zap_err_t zerr;
- again:
-	if (x->sa_len) {
-		/* use cached addresses */
-		*sa_len = x->sa_len;
-		memcpy(local_sa, &x->local_sa, x->sa_len);
-		memcpy(remote_sa, &x->remote_sa, x->sa_len);
-		return 0;
-	}
-	zerr = zap_get_name(x->zap_ep, &x->local_sa, &x->remote_sa, &x->sa_len);
-	if (zerr) {
-		x->sa_len = 0;
-		return -1;
+	socklen_t __sa_len = *sa_len;
+
+	if (!x->sa_len) {
+		zerr = zap_get_name(x->zap_ep, &x->local_sa,
+					&x->remote_sa, &__sa_len);
```

Comment:
@tom95858 Here `ldms_xprt_sockaddr` passes the `*sa_len` value to `zap_get_name`.